pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -105,6 +105,15 @@ func extractTopologyFromLabels(labels map[string]string) (map[string]string, str
 		topology["host"] = host
 	}
 
+	// ignore the region k8s topology label when it has the same value as the k8s host label
+	if topology["host"] == topology[regionLabel] {
+		delete(topology, regionLabel)
+	}
+	// ignore the zone k8s topology label when it has the same value as the k8s host label
+	if topology["host"] == topology[zoneLabel] {
+		delete(topology, zoneLabel)
+	}
+
```

Comment:
How is this addition different from the addition below? It seems to me that the addition below is more ideal since it should handle all duplicates, not just host? Or am I missing something?

Reply:
Why are we managing the duplication in two places using two methods in the code? We manage "host" topology before getting CRUSH map labels, but we manage the rest of the topology after getting the CRUSH map labels. Unless there is no other way, we should keep items of code that are logically related close to one another.

Reply:
Yaa agree